packages/realm_dart/src/realm_dart_sync.cpp

```diff
@@ -233,8 +233,10 @@ RLM_API bool realm_dart_sync_after_reset_handler_callback(realm_userdata_t userd
 RLM_API void realm_dart_async_open_task_callback(realm_userdata_t userdata, realm_thread_safe_reference_t* realm, const realm_async_error_t* error)
 {
     auto ud = reinterpret_cast<realm_dart_userdata_async_t>(userdata);
-    ud->scheduler->invoke([ud, realm, error]() {
-        (reinterpret_cast<realm_async_open_task_completion_func_t>(ud->dart_callback))(ud->handle, realm, error);
+    realm_async_error_t* error_clone = error ? error->clone() : nullptr;
```

Comment:
This was a pre-existing bug that I noticed - the passed in `error` pointer gets destroyed immediately after we exit this function, which means that when the scheduler executes the callback, it's no longer pointing to anything meaningful.